src/test/java/com/coherentsolutions/aqa/java/api/makarevich/CreateUserTest.java

```diff
@@ -0,0 +1,57 @@
+package com.coherentsolutions.aqa.java.api.makarevich;
+
+import com.coherentsolutions.aqa.java.api.makarevich.constants.UserSex;
+import com.coherentsolutions.aqa.java.api.makarevich.model.User;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+
+import static org.apache.http.HttpStatus.*;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class CreateUserTest extends TestBase {
+    @Test
+    public void createUserWithAllFieldsTest() {
+        String zipcode = zipCodeService.getRandomZipCode();
+        zipCodeService.addZipCode(zipcode);
+        ArrayList<String> zipCodes = zipCodeService.getZipCodes();
+        assertTrue(zipCodes.contains(zipcode), "Zip code is not added");
+
+        User user = userService.createRandomUser(zipcode);
+        userService.createUser(user, SC_CREATED);
+        assertTrue(userService.isUserAdded(user), "User is not added");
+
+        ArrayList<String> zipCodesAfter = zipCodeService.getZipCodes();
+        assertFalse(zipCodesAfter.contains(zipcode), "Zip code is not removed");
```

Comment:
No need to verify data from each step. For test that check user creation you should have validations is user was created correctly. You expect that other calls are working correctly. If not, there are another tests that should catch it.

Reply:
Deleted the verification for additional steps, left only verification for the main check for both CreateUserTest and ZipCodeTest